transport/src/main/java/io/camunda/zeebe/transport/impl/AtomixServerTransport.java

```diff
@@ -78,29 +76,27 @@ public void close() {
   }
 
   @Override
-  public ActorFuture<Void> unsubscribe(final int partitionId) {
+  public ActorFuture<Void> unsubscribe(final int partitionId, final RequestType requestType) {
     return actor.call(() -> removePartition(partitionId));
   }
 
   private void removePartition(final int partitionId) {
     // to unsubscribe from the partition, we can simply unsubscribe from all possible request types
     // for that partition id, even if we're not yet subscribed to some
     Arrays.stream(RequestType.values())
-        .forEach(
-            requestType -> {
-              final var topicName = topicName(partitionId, requestType);
-              if (LOG.isTraceEnabled()) {
-                LOG.trace("Unsubscribe from topic {}", topicName);
-              }
-              messagingService.unregisterHandler(topicName);
-            });
-
+        .forEach(requestType -> removeRequestHandlers(partitionId, requestType));
```

Comment:
Here you are removing all subscriptions including command, query and admin requests. Shouldn't you be only unsubscribe from specified requestType?

Reply:
Ugh, thanks for catching that. I've adjusted `unsubscribe` to use `removeRequestHandlers` which unsubscribes for one type specifically. The `removePartition` method still removes all handlers because it's called when the server transport is closed.